HotelReservation/HotelReservation/ReservationResult.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using HotelReservation.ReservationSteps;
+using System.Collections.ObjectModel;
+
+namespace HotelReservation
```

Comment:
Jest tutaj pewne niebezpieczeństwo
IsSuccessful ustwaić na true i dodać incorrect input types.
Czasami warto uzależnić jedno od drugiego np. jeśli masz kolekcję Errors, mozesz zrobić IsSuccessful => Errors.Any().

Ponieważ chcesz zrobić jeszcze API to przymknijmy na to oko :)

Reply:
Z drugiej strony jeśli u Ciebie isSuccessful ma być  false tylko na podstawie incorrect inputs, możesz tę logikę zawrzeć tutaj, a nie w każdym stepie ;)

Reply:
Fixed